pkg/fileutil/io.go

```diff
@@ -131,3 +136,34 @@ func FileExists(p string) (bool, error) {
 	}
 	return false, fmt.Errorf("%s: %w", p, ErrNotFile)
 }
+
+func VerifyAbsPath(absPath, basePath string) error {
+	// check we have a valid abs path
+	if !path.IsAbs(absPath) || path.Clean(absPath) != absPath {
+		return ErrBadPath
+	}
+	// point to storage namespace
+	if !strings.HasPrefix(absPath, basePath) {
+		return ErrInvalidPath
+	}
+	return nil
+}
+
+func VerifyRelPath(relPath, base string) error {
+	abs := base + string(os.PathSeparator) + relPath
+	return VerifyAbsPath(abs, base)
+}
+
+func VerifyRegularFile(name, base string) error {
+	filename, err := filepath.EvalSymlinks(name)
+	switch {
+	case err != nil:
+		return err
+	case filename != name:
+		return ErrRegularFile
```

Comment:
Isn't this `ErrBadPath`?

Reply:
This will be confusing to the user - they will not understand why this is a bad path. This error should make it clear